homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,806 @@
+            except (asyncio.TimeoutError, aiohttp.ClientError):
+                # Not yet seen alive, leave for now, gracefully
+                return
+
+        # call GetTransportInfo/GetPositionInfo regularly
+        try:
+            avt_service = self._service('AVT')
+            if avt_service:
+                await self._async_poll_transport_info(avt_service)
+                if self.state == STATE_PLAYING or \
+                   self.state == STATE_PAUSED:
+                    # playing something... get position info
+                    await self._async_poll_position_info(avt_service)
+            else:
+                await self._device.async_ping()
+
+            self._is_connected = True
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error('Error during update call')
+            self._is_connected = False
+            await self.async_unsubscribe_all()
+
+    async def _async_poll_transport_info(self, avt_service):
+        """Update transport info from device."""
+        action = avt_service.action('GetTransportInfo')
+        result = await action.async_call(InstanceID=0)
+
+        # set/update state_variable 'TransportState'
+        changed = []
+        service = action.service
+        state_var = service.state_variable('TransportState')
+        if state_var.value != result['CurrentTransportState']:
+            state_var.value = result['CurrentTransportState']
+            changed.append(state_var)
+
+        self.on_state_variable_change(service, changed)
+
+    async def _async_poll_position_info(self, avt_service):
+        """Update position info."""
+        action = avt_service.action('GetPositionInfo')
+        result = await action.async_call(InstanceID=0)
+
+        changed = []
+        service = action.service
+        track_duration = service.state_variable('CurrentTrackDuration')
+        if track_duration.value != result['TrackDuration']:
+            track_duration.value = result['TrackDuration']
+            changed.append(track_duration)
+
+        time_position = service.state_variable('RelativeTimePosition')
+        if time_position.value != result['RelTime']:
+            time_position.value = result['RelTime']
+            changed.append(time_position)
+
+        self.on_state_variable_change(service, changed)
+
+    def on_state_variable_change(self, service, state_variables):
+        """State variable(s) changed, let home-assistant know."""
+        for state_variable in state_variables:
+            if state_variable.name == 'LastChange':
+                from async_upnp_client.utils import \
+                    dlna_handle_notify_last_change
+                dlna_handle_notify_last_change(state_variable)
+
+        if state_variables:
+            self.schedule_update_ha_state()
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return SUPPORT_DLNA_DMR
```

Comment:
Is it possible to induce what features are supported based on api calls? If yes, I think that information should be gathered before the device entity is initialized to avoid "broken"/missing functionality.

Reply:
This implies not being able to initialize a device without it being on. I.e., during a (re)start of hass the device (e.g., TV) must be on, otherwise it cannot be initialized when manually configured. Discovery is another story of course, then the device is on/accessible.

`media_player.mpd` had this behavior before and was fairly annoying. It was fixed by #8655. Made at least @pezinek happy. :)

`supported_features` was dynamic before, but was made static due to architectural requirements.

Reply:
> This implies not being able to initialize a device without it being on. I.e., during a (re)start of hass the device (e.g., TV) must be on, otherwise it cannot be initialized when manually configured.

The platform can raise `PlatformNotReady` exception and homeassistant will keep trying to initialize it later (uses a backoff timer for that), which is to my understanding the correct way to handle that.